radio/src/targets/horus/adc_driver.cpp

```diff
@@ -126,103 +124,118 @@ void adcInit()
-  DMA2_Stream1->M0AR = CONVERT_PTR_UINT(&adcValues[MOUSE1]);
-  DMA2_Stream1->NDTR = 2;
-  DMA2_Stream1->CR |= DMA_SxCR_EN;		// Enable DMA
-  ADC3->CR2 |= (uint32_t)ADC_CR2_SWSTART;
-
-  // A dummy command to get things started 
+  // A dummy command to get things started
   // (because the sampled data is lagging behind for two command cycles)
   ADC_CS_LOW();
   delay_01us(1);
-  SPIx_ReadWriteByte(*command);   // still sampling the old MUX channel
+  SPIx_ReadWriteByte(adcCommands[0]);
   ADC_CS_HIGH();
   delay_01us(1);
+}
 
-  DEBUG_TIMER_START(debugTimerAdcLoop);
-  for (uint32_t adcIndex=0; adcIndex<MOUSE1; adcIndex++) {
-    // MUX is changed to the channel that was sent in the previous command
-    // but the data is from the old MUX position
-    ADC_CS_LOW();
-    delay_01us(1);
-    SPIx_ReadWriteByte(*command);   
-    ADC_CS_HIGH();
-    delay_01us(1);
+uint32_t adcReadNextSPIChannel(uint8_t index)
+{
+  uint32_t result = 0;
 
-    // This delay is to allow charging of ADC input capacitor 
-    // after the MUX changes from one channel to the other.
-    // It was determined experimentally. Biggest problem seems to be 
-    // the cross-talk between A4:S1 and A5:MULTIPOS. Changing S1 for one extreme
-    // to the other resulted in A5 change of:
-    //
-    //        delay value       A5 change
-    //          0               76 
-    //        100               26
-    //        300               13
-    //        500               undetectable
-    delay_01us(500);
-    ++command;    // move to the next channel
-
-    // First time the wanted ADC channel is actually sampled.
-    // This is delayed to allow MUX to settle (high input impedance)
-    // Also the MUX channel is set to the next channel here, 
-    // but the actual MUX change will happen in the next round
+  ADC_CS_LOW();
+  delay_01us(1);
+  SPIx_ReadWriteByte(adcCommands[index]);
+  ADC_CS_HIGH();
+  // This delay is to allow charging of ADC input capacitor
+  // after the MUX changes from one channel to the other.
+  // It was determined experimentally. Biggest problem seems to be
+  // the cross-talk between A4:S1 and A5:MULTIPOS. Changing S1 for one extreme
+  // to the other resulted in A5 change of:
+  //
+  //        delay value       A5 change
+  //          0               76
+  //        100               26
+  //        300               13
+  //        500               undetectable
+  delay_01us(500);
+
+  for (uint8_t i = 0; i < 4; i++) {
     ADC_CS_LOW();
     delay_01us(1);
-    adcValues[adcIndex] = (0x0fff & SPIx_ReadWriteByte(*command));
+    uint16_t val = (0x0fff & SPIx_ReadWriteByte(adcCommands[index]));
+#if defined(JITTER_MEASURE)
+    if (JITTER_MEASURE_ACTIVE()) {
+      rawJitter[index].measure(val);
+    }
+#endif
     ADC_CS_HIGH();
     delay_01us(1);
+    result += val;
   }
-  DEBUG_TIMER_STOP(debugTimerAdcLoop);
 
-  DEBUG_TIMER_START(debugTimerAdcWait);
+  return result >> 2;

```

Comment:
I would prefer if `result` was not divided by 4. We have enough bits in uint16_t to store four full ADC readings (4 * 0xFFF = 3FFC). Dividing here means lost precision/resolution for the jitter filter. So